lte/gateway/c/core/oai/test/mme_app_task/test_mme_procedures.cpp

```diff
@@ -2886,5 +2913,222 @@ TEST_F(MmeAppProcedureTest, TestTauRejDueToInvalidTac) {
+
+  // Sending AIA to mme_app mimicing successful S6A response for AIR
+  send_authentication_info_resp(imsi, true);
+
+  // Wait for DL NAS Transport for once
+  cv.wait_for(lock, std::chrono::milliseconds(STATE_MAX_WAIT_MS));
+  // Constructing and sending Authentication Response to mme_app mimicing S1AP
+  send_mme_app_uplink_data_ind(
+      nas_msg_auth_resp, sizeof(nas_msg_auth_resp), plmn);
+
+  // Wait for DL NAS Transport for once
+  cv.wait_for(lock, std::chrono::milliseconds(STATE_MAX_WAIT_MS));
+  // Constructing and sending SMC Response to mme_app mimicing S1AP
+  send_mme_app_uplink_data_ind(
+      nas_msg_smc_resp, sizeof(nas_msg_smc_resp), plmn);
+
+  // Sending ULA to mme_app mimicing successful S6A response for ULR
+  send_s6a_ula(imsi, true);
+
+  // Constructing and sending Create Session Response to mme_app mimicing SPGW
+  send_create_session_resp(REQUEST_ACCEPTED);
+
+  // Constructing and sending ICS Response to mme_app mimicing S1AP
+  send_ics_response();
+
+  // Constructing UE Capability Indication message to mme_app
+  // mimicing S1AP with dummy radio capabilities
+  send_ue_capabilities_ind();
+
+  // Constructing and sending Attach Complete to mme_app
+  // mimicing S1AP
+  send_mme_app_uplink_data_ind(
+      nas_msg_attach_comp, sizeof(nas_msg_attach_comp), plmn);
+
+  // Wait for DL NAS Transport for EMM Information
+  cv.wait_for(lock, std::chrono::milliseconds(STATE_MAX_WAIT_MS));
+
+  // Constructing and sending Modify Bearer Response to mme_app
+  // mimicing SPGW
+  std::vector<int> b_modify = {5};
+  std::vector<int> b_rm     = {};
+  send_modify_bearer_resp(b_modify, b_rm);
+
+  // Check MME state after Modify Bearer Response
+  send_activate_message_to_mme_app();
+  cv.wait_for(lock, std::chrono::milliseconds(STATE_MAX_WAIT_MS));
+  EXPECT_EQ(mme_state_p->nb_ue_attached, 1);
+  EXPECT_EQ(mme_state_p->nb_ue_connected, 1);
+  EXPECT_EQ(mme_state_p->nb_default_eps_bearers, 1);
+  EXPECT_EQ(mme_state_p->nb_ue_idle, 0);
+
+  // Send path switch request to mme_app mimicing S1AP, use 1 for all new ids,
+  // since Initial UE message is using 0
+  uint32_t new_enb_ue_s1ap_id = DEFAULT_eNB_S1AP_UE_ID + 1;
+  uint32_t new_sctp_assoc_id  = DEFAULT_SCTP_ASSOC_ID + 1;
+  uint32_t new_enb_id         = DEFAULT_ENB_ID + 1;
+  send_s1ap_path_switch_req(
+      new_sctp_assoc_id, new_enb_id, new_enb_ue_s1ap_id, plmn);
+
+  // Expect that Path Switch Failure is sent to S1AP from mme_app after modify
+  // bearer response is received
+  EXPECT_CALL(
+      *s1ap_handler, s1ap_handle_path_switch_req_failure(
+                         check_params_in_path_switch_req_failure(
+                             new_enb_ue_s1ap_id, new_sctp_assoc_id)))
+      .Times(1);
+
+  // Constructing and sending Modify Bearer Response to mme_app
+  // mimicing SPGW, with empty modify bearer list to trigger failure
+  b_modify = {};
+  send_modify_bearer_resp(b_modify, b_rm);
```

Comment:
shall we check the expected state after MBR response?

Reply:
Check added